admin/services/data-access/pupil-age-reason.data.service.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const { TYPES } = require('./sql.service')
+
+const sqlService = require('./sql.service')
+const pupilAgeReasonDataService = {}
+
+/**
+ * Insert pupil age reason record and link with pupil age reason id
+ * @param {Number} pupilId
+ * @param {String} reason
+ * @returns {Promise<Array>}
+ */
+pupilAgeReasonDataService.sqlInsertPupilAgeReason = async function (pupilId, reason) {
+  const sql = `
+  INSERT INTO ${sqlService.adminSchema}.[pupilAgeReason]
+  (pupil_id, reason)
+  VALUES (@pupilId, @reason)
+  
+  UPDATE ${sqlService.adminSchema}.[pupil]
+  SET pupilAgeReason_id = SCOPE_IDENTITY()
+  WHERE id = @pupilId
+  `
+
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    },
+    {
+      name: 'reason',
+      value: reason,
+      type: TYPES.NVarChar
+    }
+  ]
+
+  return sqlService.modify(sql, params)
```

Comment:
As there are actually two SQL statements here it would be preferable to wrap this in a transaction to ensure atomicity.  Would you consider changing this to use `sqlService.modifyWithTransaction()`?

Reply:
✅